### PR TITLE
refactor: Replace string-based import parsing with AST-based detection

### PR DESCRIPTION
**Pull request by StefanoFioravanzo**
## Summary
- Introduce a new `imports.py` module with proper AST-based import parsing
- Replace brittle hardcoded string matching in the compiler
- Add comprehensive test coverage (70 unit tests)

## Problem

When Kale compiles a notebook into a Kubeflow Pipeline, it needs to determine which Python packages should be installed in the pipeline steps. This is done by analyzing the import statements in the notebook cells.

The previous implementation used simple string matching to parse imports:

https://github.com/kubeflow/kale/blob/5c4a324ed915eed6d0a279622a8bb28f381a8453/backend/kale/compiler.py#L279-L298

This approach had several issues:

1. Hardcoded mappings: Only `random` and `sklearn` were mapped to their PyPI names. Other common mismatches (e.g., `PIL` → `pillow`, `cv2` → `opencv-python`, `yaml` → `pyyaml`) were not handled.
2. No stdlib filtering: Standard library modules like `os`, `sys`, `json`, etc. were being added to the package list unnecessarily.
3. Fragile parsing: String splitting doesn't handle all valid Python import syntaxes:

  - `import a, b, c` (multiple imports)
  - `from x import (a, b, c)` (parenthesized imports)
  - Multi-line imports
  - Comments on import lines
4. Not extensible: Adding new package mappings required modifying the compiler code directly.

## Solution
This PR introduces a new `backend/kale/common/imports.py` module that uses Python's `ast` module to properly parse imports:

**Key Components**
- `STDLIB_MODULES`: A comprehensive set of Python 3.12+ standard library modules to exclude from package requirements.
- `PACKAGE_NAME_MAP`: A centralized registry mapping import names to PyPI package names:

```python
PACKAGE_NAME_MAP = {
    "sklearn": "scikit-learn",
    "cv2": "opencv-python",
    "PIL": "pillow",
    "yaml": "pyyaml",
    # ... and more
}
```

- `ImportInfo:` A dataclass providing structured information about each import (module, names, alias, line number).
- `parse_imports_ast()`: Parses all imports from source code using AST, handling all valid Python import syntaxes.
- `get_packages_to_install()`: Returns the set of PyPI packages needed for given source code, filtering stdlib and applying name mappings.

**Benefits**
- Correct parsing: Handles all valid Python import syntax
- Extensible: New package mappings can be added to PACKAGE_NAME_MAP
- Maintainable: Centralized configuration instead of scattered conditionals
- Tested: 70 unit tests covering all import forms and edge cases

**Changes**
- New module: `backend/kale/common/imports.py`
- Refactored: `backend/kale/compiler.py` - Use new module instead of string matching
- Tests: `backend/kale/tests/unit_tests/test_imports.py` - 70 comprehensive tests

### CODE REVIEW

File: backend/kale/compiler.py [hunk 1]
```diff
@@ -260,42 +261,29 @@ def generate_pipeline(self, lightweight_components):
...
-                    if package_name == "sklearn":
-                        package_name = "scikit-learn"
-                    package_names.add(package_name)
-        return sorted(package_names)
+
+        # Parse imports using AST and resolve to PyPI package names
+        package_names.update(get_packages_to_install(self.imports_and_functions))
+
+        return sorted(list(package_names))
+>>>>>>> 3e0ed2d (refactor: Use AST-based import parsing in compiler)
```
Comment (jesuino) [inline backend/kale/compiler.py]: It looks like a left over of a rebase/merge!

Comment (StefanoFioravanzo) [reply backend/kale/compiler.py]: Thanks! should be fixed now

**Comment by FAUST-BENCHOU** [comment]
so many lint error
`uv run ruff check backend --fix` may fix them?

**Comment by StefanoFioravanzo** [comment]
@FAUST-BENCHOU thanks, all checks are passing now!

**Comment by StefanoFioravanzo** [comment]
@ederign can you also please review and +1?
